### ISSUE TITLE
measureList and measureSelector in APS

### ISSUE THREAD

**Issue by franckblais** [org-scn-design-studio-community/sdkpackage#120]
Hi !

I was looking at the different controls in the APS and found measureList and stuff.
I am interested by using it ... but I don't understand how to give him the propertyPage.flatData variable ...

Could you help me understand ?

Thanks !

**Comment by entmike** [comment]
To use those, yes your component must support this flatData approach.  This is specified in the getComponentInfo call performed on all components here:

https://github.com/org-scn-design-studio-community/sdkpackage/blob/master/src/org.scn.community.shared/aps/PropertyPage.js#L319 

and if component info is returned, a 'supportsFlatdata' check is done here:

https://github.com/org-scn-design-studio-community/sdkpackage/blob/master/src/org.scn.community.shared/aps/PropertyPage.js#L401

For an example of how to specify getComponentInfo function, you can either use Karol's generation approach to specifiy it, or via callRuntimehandler.  Whichever approach you decide, this is the general format to return via function or via @KarolKalisz method.

Mine is a bit hard to follow due to my inheritance pattern but see these properties to return via function:

https://github.com/org-scn-design-studio-community/sdkpackage/blob/153ed11f90521260a87ff86e98646404f568af9f/src/org.scn.community.shared/os/viz-modules/VizCoreDatabound.js#L117-L130

It is inherited here:

https://github.com/org-scn-design-studio-community/sdkpackage/blob/153ed11f90521260a87ff86e98646404f568af9f/src/org.scn.community.shared/os/viz-modules/SDKCore.js#L19-L32

The simple function is here:

https://github.com/org-scn-design-studio-community/sdkpackage/blob/153ed11f90521260a87ff86e98646404f568af9f/src/org.scn.community.shared/os/viz-modules/SDKCore.js#L82-L84

Basically, just relay your componentInfo JSON to include that flag.  You will also then have to support that flattened data.  Again, Karol does this one way, I do it another.  But you can see the flatten function at work here:

https://github.com/org-scn-design-studio-community/sdkpackage/blob/153ed11f90521260a87ff86e98646404f568af9f/src/org.scn.community.shared/os/viz-modules/VizCoreDatabound.js#L14-L33

So have this available in your code in order to set a flatData variable.

Once you have your info set, the APS will know to do another call to help the measureList stuff via another 2 callRuntimeHandler calls you can see here:

https://github.com/org-scn-design-studio-community/sdkpackage/blob/153ed11f90521260a87ff86e98646404f568af9f/src/org.scn.community.shared/os/viz-modules/VizCoreDatabound.js#L131-L166

With these things in place, these special APS components you saw will work automagically.

Let me know if my response is too weird or you need me to clarify!

**Comment by franckblais** [comment]
Thanks Mike ! I'll have a look at it. I had quite a hard time figuring where it was used :)
